File: algoexpert/easy/javascript/Binary_search.js
const BinarySearch = (nums, target) => {
	let l = 0;
	let h = nums.length;
	while (l < h) {
		let mid = Math.floor((l + h) / 2);
		if (nums[mid] === target)
			return mid;
		else if (nums[mid] > target)
			h = mid;
		else
			l = mid + 1;
	}
	return -1;
}

console.log(BinarySearch([0, 1, 21, 33, 45, 45, 61, 71, 72, 73], 33));

File: algoexpert/easy/python/Binary_search.py
def BinarySearch(nums, n):
	l = 0
	h = len(nums)
	while l < h:
		mid = (l + h) // 2
		if nums[mid] == n:
			return mid
		elif nums[mid] > n:
			h = mid
		else:
			l = mid + 1
	return -1


print(BinarySearch([0, 1, 21, 33, 45, 45, 61, 71, 72, 73], 33))
